packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -748,90 +820,494 @@ export class Domain extends cdk.Resource implements IDomain {
+        policyName: 'ESLogPolicy',
+        policyStatements: [logPolicyStatement],
+      });
+    }
+
+    // Create the domain
+    this.domain = new CfnDomain(this, 'Resource', {
+      domainName: this.physicalName,
+      elasticsearchVersion: elasticsearchVersion,
+      elasticsearchClusterConfig: {
+        dedicatedMasterEnabled: props.clusterConfig.masterNodes != null,
+        dedicatedMasterCount: props.clusterConfig.masterNodes,
+        dedicatedMasterType: props.clusterConfig.masterNodeInstanceType,
+        instanceCount: props.clusterConfig.dataNodes,
+        instanceType: props.clusterConfig.dataNodeInstanceType,
+        zoneAwarenessEnabled: props.clusterConfig.availabilityZoneCount != null,
+        zoneAwarenessConfig:
+          props.clusterConfig.availabilityZoneCount != null
+            ? { availabilityZoneCount: props.clusterConfig.availabilityZoneCount }
+            : undefined,
+      },
+      ebsOptions: {
+        ebsEnabled,
+        volumeSize: props.ebsOptions?.volumeSize,
+        volumeType: props.ebsOptions?.volumeType,
+        iops: props.ebsOptions?.iops,
+      },
+      encryptionAtRestOptions: {
+        enabled: encryptionAtRestEnabled,
+        kmsKeyId: props.encryptionAtRestOptions?.kmsKey?.keyId,
+      },
+      nodeToNodeEncryptionOptions: { enabled: props.nodeToNodeEncryptionEnabled ?? false },
+      logPublishingOptions: {
+        ES_APPLICATION_LOGS: {
+          enabled: this.appLogGroup != null,
+          cloudWatchLogsLogGroupArn: this.appLogGroup?.logGroupArn,
+        },
+        SEARCH_SLOW_LOGS: {
+          enabled: this.slowSearchLogGroup != null,
+          cloudWatchLogsLogGroupArn: this.slowSearchLogGroup?.logGroupArn,
+        },
+        INDEX_SLOW_LOGS: {
+          enabled: this.slowIndexLogGroup != null,
+          cloudWatchLogsLogGroupArn: this.slowIndexLogGroup?.logGroupArn,
+        },
+      },
+      cognitoOptions: {
+        enabled: props.cognitoOptions != null,
+        identityPoolId: props.cognitoOptions?.identityPoolId,
+        roleArn: props.cognitoOptions?.role.roleArn,
+        userPoolId: props.cognitoOptions?.userPoolId,
+      },
+      vpcOptions: cfnVpcOptions,
+      snapshotOptions: props.automatedSnapshotStartHour
+        ? { automatedSnapshotStartHour: props.automatedSnapshotStartHour }
+        : undefined,
+    });
+
+    if (logGroupResourcePolicy) { this.domain.node.addDependency(logGroupResourcePolicy); }
+
+    if (props.domainName) { this.node.addMetadata('aws:cdk:hasPhysicalName', props.domainName); }
+
+    this.encryptionKey = props.encryptionAtRestOptions?.kmsKey;
+    this.domainArn = this.getResourceArnAttribute(this.domain.attrArn, {
+      service: 'es',
+      resource: 'domain',
+      resourceName: this.physicalName,
+    });
+    this.domainName = this.getResourceNameAttribute(this.domain.ref);
+
+    this.domainEndpoint = this.domain.getAtt('DomainEndpoint').toString();
+  }
+}
+
+/**
```

Comment:
Maybe it would be better just to go back to this value as a string; we could validate that 1) it's a number and 2) that any number less than the current latest (7.4) matches a known good version; we don't try to validate anything greater than current latest. Thoughts?

Reply:
cc @stephanh ☝️

Reply:
I went ahead with this approach; let me know what you think! af39b84

Reply:
Sorry for not thinking about this before but my concern now is that in the future AWS ES might have a version string that is not just major.minor.

I think we can leave that for a future change though.

Reply:
I'm happy to change it back to a string if that's preferable!

Reply:
I was planning on finishing that off. I'm thinking off:

1. Change it to string.
1. Try to parse the `x.y` prefix.
1. Provide string constants for the current versions.

Reply:
I made the changes if you want to have a look 3bd4b637760986213efba5f134bb78217c8f3b84.